quickwit/quickwit-ingest-api/src/codegen/ingest_service.rs

```diff
@@ -227,6 +230,138 @@ impl tower::Service<TailRequest> for IngestServiceClient {
+        IngestRequest,
+        IngestResponse,
+        crate::IngestServiceError,
+    >,
+    fetch_svc: quickwit_common::tower::BoxService<
+        FetchRequest,
+        FetchResponse,
+        crate::IngestServiceError,
+    >,
+    tail_svc: quickwit_common::tower::BoxService<
+        TailRequest,
+        FetchResponse,
+        crate::IngestServiceError,
+    >,
+}
+impl Clone for IngestServiceTowerBlock {
+    fn clone(&self) -> Self {
+        Self {
+            ingest_svc: self.ingest_svc.clone(),
+            fetch_svc: self.fetch_svc.clone(),
+            tail_svc: self.tail_svc.clone(),
+        }
+    }
+}
+#[async_trait::async_trait]
+impl IngestService for IngestServiceTowerBlock {
+    async fn ingest(&mut self, request: IngestRequest) -> crate::Result<IngestResponse> {
+        self.ingest_svc.ready().await?.call(request).await
+    }
+    async fn fetch(&mut self, request: FetchRequest) -> crate::Result<FetchResponse> {
+        self.fetch_svc.ready().await?.call(request).await
+    }
+    async fn tail(&mut self, request: TailRequest) -> crate::Result<FetchResponse> {
+        self.tail_svc.ready().await?.call(request).await
+    }
+}
+#[derive(Debug, Default)]
+pub struct IngestServiceTowerBlockBuilder {
+    ingest_layer: Option<
+        quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            IngestRequest,
+            IngestResponse,
+            crate::IngestServiceError,
+        >,
+    >,
+    fetch_layer: Option<
+        quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            FetchRequest,
+            FetchResponse,
+            crate::IngestServiceError,
+        >,
+    >,
+    tail_layer: Option<
+        quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            TailRequest,
+            FetchResponse,
+            crate::IngestServiceError,
+        >,
+    >,
+}
+impl IngestServiceTowerBlockBuilder {
+    pub fn ingest_layer(
+        mut self,
+        layer: quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            IngestRequest,
+            IngestResponse,
+            crate::IngestServiceError,
+        >,
+    ) -> Self {
+        self.ingest_layer = Some(layer);
+        self
+    }
+    pub fn fetch_layer(
+        mut self,
+        layer: quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            FetchRequest,
+            FetchResponse,
+            crate::IngestServiceError,
+        >,
+    ) -> Self {
+        self.fetch_layer = Some(layer);
+        self
+    }
+    pub fn tail_layer(
+        mut self,
+        layer: quickwit_common::tower::BoxLayer<
+            Box<dyn IngestService>,
+            TailRequest,
+            FetchResponse,
+            crate::IngestServiceError,
+        >,
+    ) -> Self {
+        self.tail_layer = Some(layer);
+        self
+    }
+    pub fn service<T>(self, instance: T) -> IngestServiceClient
```

Comment:
`build` instead of `searvice`?

Reply:
I hesitated on that one. I'm mimicking the Tower API, which uses `.layer().layer().service()`.